### PR TITLE
extension store guide

### PR DESCRIPTION
**Pull request by jim-docker**
Signed-off-by: Jim Ehrismann <jehrismann@mirantis.com>

nearly completes #1240

### CODE REVIEW

File: docs/extensions/guides/stores.md [hunk 1]
```diff
@@ -1,11 +1,158 @@
...
+
+Then the constructor and two abstract methods are implemented.
+In the constructor, the name of the store (`"example-preferences-store"`), and the default (initial) value for the preference state (`enabled: false`) are specified. 
+The `fromStore()` method is called by Lens internals when the store is loaded, and gives the extension the opportunity to retrieve the stored state data values based on the defined data model.
+Here, the `enabled` field of the `ExamplePreferencesStore` is set to the value from the store whenever `fromStore()` is invoked.
+The `toJSON()` method is complementary to `fromStore()`, and is called when the store is being saved.
+`toJSON()` must provide a JSON serializable object, facilitating its storage in JSON format.
+The `toJS()` function from [`mobx`](https://mobx.js.org/README.html) is convenient for this purpose, and is used here.
+
+Finally, the `examplePreferencesStore` is created by calling `ExamplePreferencesStore.getInstance<ExamplePreferencesStore>()`, and exported for use by other parts of the extension.
```
Comment (panuhorsmalahti) [inline docs/extensions/guides/stores.md]: We could add the word "singleton" here, e.g.
```
`examplePreferencesStore` singleton
```

File: docs/extensions/guides/stores.md [hunk 2]
```diff
@@ -1,11 +1,158 @@
...
+
+The following example code, modified from the [`appPreferences` guide example](../renderer-extension#apppreferences) demonstrates how to use the extension store.
+`examplePreferencesStore` must be loaded in the main process, where loaded stores are automatically saved when exiting Lens. This can be done in `./main.ts`:
```
Comment (panuhorsmalahti) [inline docs/extensions/guides/stores.md:74]: This could be clarified by e.g. "must be loaded using `loadExtension`"

**Review by Nokel81 — DISMISSED** [review]

File: docs/extensions/guides/stores.md [hunk 3]
```diff
@@ -5,13 +5,6 @@ A few of these are exposed by the extensions api for use by the extension develo
 The `ClusterStore` manages cluster state data such as cluster details, and which cluster is active.
 The `WorkspaceStore` similarly manages workspace state data, such as workspace name, and which clusters belong to a given workspace.
 The `ExtensionStore` is a store for managing custom extension state data.
-## ClusterStore
-
-tbd
-
-## WorkspaceStore
-
-tbd
```
Comment (jim-docker) [inline docs/extensions/guides/stores.md]: better to leave this vague (i.e. missing) than tbd, I think

File: docs/extensions/guides/stores.md [hunk 4]
```diff
@@ -1,11 +1,154 @@
...
----
-
 # Stores
 
-## ClusterStore
+Stores are components that persist and synchronize state data. Lens utilizes a number of stores for maintaining a variety of state information.
+A few of these are exposed by the extensions api for use by the extension developer.
+The `ClusterStore` manages cluster state data such as cluster details, and which cluster is active.
+The `WorkspaceStore` similarly manages workspace state data, such as workspace name, and which clusters belong to a given workspace.
+The `ExtensionStore` is a store for managing custom extension state data.
```
Comment (Nokel81) [inline docs/extensions/guides/stores.md]:
```suggestion
- The `ClusterStore` manages cluster state data such as cluster details, and which cluster is active.
- The `WorkspaceStore` similarly manages workspace state data, such as workspace name, and which clusters belong to a given workspace.
- The `ExtensionStore` is a store for managing custom extension state data.
```

File: docs/extensions/guides/stores.md [hunk 5]
```diff
@@ -1,11 +1,154 @@
...
+
+## ExtensionStore
+
+Extension developers can create their own store for managing state data by extending the `ExtensionStore` class.
+This guide shows how to create a store for the [`appPreferences` guide example](../renderer-extension#apppreferences), which demonstrates how to add a custom preference to the Preferences page.
+The preference is a simple boolean that indicates whether something is enabled or not.
+The problem with this example is that the enabled state is not stored anywhere, and reverts to the default the next time Lens is started.
```
Comment (Nokel81) [inline docs/extensions/guides/stores.md]:
```suggestion
The problem with that example is that the enabled state is not stored anywhere, and reverts to the default the next time Lens is started.
```

Comment (jim-docker) [reply docs/extensions/guides/stores.md]: Didn't you already approve this? :)

Comment (Nokel81) [reply docs/extensions/guides/stores.md]: Yes, just noticed these this time around

**Review by Nokel81 — APPROVED** [review]

**Review by panuhorsmalahti — APPROVED** [review]

**Pull request merged**